app/main.py

```diff
@@ -306,14 +300,30 @@ def grab_secrets(self):
                 _logger.info('Dumping cert value to: %s', output_path)
                 with open(output_path, 'w') as cert_file:
                     cert_file.write(self._cert_to_pem(cert.cer))
+    
+    def _load_cert(self, secret_value, content_type):
+        if (content_type == 'application/x-pkcs12'):
+            pkcs12 = crypto.load_pkcs12(base64.decodestring(secret_value))
+            x509 = pkcs12.get_certificate()
+            key = pkcs12.get_privatekey()
+        elif (content_type == 'application/x-pem-file'):
+            x509 = crypto.load_certificate(crypto.FILETYPE_PEM, secret_value.encode('utf-8'))
```

Comment:
This only loads the first cert from within a PEM - if the PEM contains a cert chain, this would miss that chain.
(It is unfortunate that there is no simple function built in to load the full chain and return a list of certs.)